File: src/test/java/kr/co/lunatalk/domain/member/domain/MemberTest.java
package kr.co.lunatalk.domain.member.domain;

import com.navercorp.fixturemonkey.FixtureMonkey;
import com.navercorp.fixturemonkey.api.introspector.ConstructorArbitraryIntrospector;
import com.navercorp.fixturemonkey.api.introspector.ConstructorPropertiesArbitraryIntrospector;
import jakarta.persistence.EntityManager;
import jakarta.persistence.PersistenceContext;
import kr.co.lunatalk.domain.member.repository.MemberRepository;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.annotation.Rollback;
import org.springframework.transaction.annotation.Transactional;

import java.util.NoSuchElementException;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;


@SpringBootTest
@Transactional
class MemberTest {

	FixtureMonkey fixtureMonkey;

	@BeforeEach
	void setUp() {
		fixtureMonkey = FixtureMonkey.builder()
			.objectIntrospector(ConstructorPropertiesArbitraryIntrospector.INSTANCE).build();
	}

	@Test
	void createMember_성공() {
		//given
		Profile profile = fixtureMonkey.giveMeOne(Profile.class);

		//when
		Member member = Member.createMember("user", "password", profile);

		//then
		assertNotNull(member);
		assertEquals(profile, member.getProfile());
		assertEquals(MemberStatus.NORMAL, member.getStatus());
		assertEquals(MemberRole.USER, member.getRole());
	}

	@Test
	void updateLastLoginAt_성공() {
		// given
		Member member = fixtureMonkey.giveMeOne(Member.class);

		//when
		member.updateLastLoginAt();

		//then
		assertNotNull(member.getLastLoginAt());
	}

	@Test
	void updateProfile() {
		// given
		Member member = fixtureMonkey.giveMeOne(Member.class);
		Profile newProfile = fixtureMonkey.giveMeOne(Profile.class);
		// when
		member.updateProfile(newProfile);
		// then
		assertEquals(newProfile, member.getProfile());
	}


}

File: src/test/kotlin/kr/co/lunatalk/domain/member/domain/MemberTest.kt
package kr.co.lunatalk.domain.member.domain

import com.navercorp.fixturemonkey.FixtureMonkey
import com.navercorp.fixturemonkey.api.introspector.ConstructorPropertiesArbitraryIntrospector
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles
import org.springframework.transaction.annotation.Transactional

@SpringBootTest
@Transactional
@ActiveProfiles("test")
class MemberTest {

    private lateinit var fixtureMonkey: FixtureMonkey

    @BeforeEach
    fun setUp() {
        fixtureMonkey = FixtureMonkey.builder()
            .objectIntrospector(ConstructorPropertiesArbitraryIntrospector.INSTANCE)
            .build()
    }

    @Test
    fun `createMember 성공`() {
        // given
        val profile = fixtureMonkey.giveMeOne(Profile::class.java)

        // when
        val member = Member.createMember(
            "user", "password", profile,
            "01012341234", "kkwondev@gmail.com"
        )

        // then
        assertNotNull(member)
        assertEquals(profile, member.profile)
        assertEquals(MemberStatus.NORMAL, member.status)
        assertEquals(MemberRole.USER, member.role)
    }

    @Test
    fun `updateLastLoginAt 성공`() {
        // given
        val member = fixtureMonkey.giveMeOne(Member::class.java)

        // when
        member.updateLastLoginAt()

        // then
        assertNotNull(member.lastLoginAt)
    }

    @Test
    fun updateProfile() {
        // given
        val member = fixtureMonkey.giveMeOne(Member::class.java)
        val newProfile = fixtureMonkey.giveMeOne(Profile::class.java)

        // when
        member.updateProfile(newProfile)

        // then
        assertEquals(newProfile, member.profile)
    }
}
